### PR TITLE
Improve support for missingness in Filters/Smoothers

### PR DESCRIPTION
**Pull request by mattlevine22**
This PR adds early missing-observation validation for filtering and smoothing, documents missingness support directly on the filter/smoother config classes, and stabilizes the continuous-time hierarchical simulator/discretizer smoke fixtures so they no longer generate accidental `NaN` rollouts. Unsupported missing-observation paths now fail early with clear messages: cuthbert `KFConfig`, `EnKFConfig`, and `KFSmootherConfig` remain the supported discrete-time `NaN` paths, while cuthbert `EKF`/`EKFSmoother` and CD-Dynamax filter/smoother paths are rejected cleanly.

It also adds missing-observation regression coverage in [tests/test_missing_observations.py](tests/test_missing_observations.py), updates the filter/smoother config docs in [dynestyx/inference/filter_configs.py](dynestyx/inference/filter_configs.py) and [dynestyx/inference/smoother_configs.py](dynestyx/inference/smoother_configs.py) to state whether each backend supports missing observations via `NaN`s, and includes the new [11_missing_observations.ipynb](docs/tutorials/gentle_intro/11_missing_observations.ipynb) tutorial in the docs navigation via [mkdocs.yml](mkdocs.yml).

### CODE REVIEW

File: dynestyx/inference/checkers.py [hunk 1]
```diff
@@ -1,13 +1,48 @@
...
+    """Raise plainly when possible; fall back to eqx.error_if under tracing."""
+    try:
+        missing_now = bool(has_missing)
+    except jax.errors.TracerBoolConversionError:
+        _ = eqx.error_if(obs_values, has_missing, message)
+        return
+
+    if missing_now:
+        raise ValueError(message)
+
```
Comment (DanWaxman) [inline dynestyx/inference/checkers.py]: This feels weird as a separate helper function. I understand it does the tracing thing, but that is much more general as helper.

File: dynestyx/inference/filter_configs.py [hunk 2]
```diff
@@ -230,6 +232,8 @@ class PFConfig(BaseFilterConfig):
     `ess_threshold_ratio` controls the frequency of resampling; sampling more frequently
     can help avoid particle degeneracy, but also increases variance.
 
+    Does not support missing observations (data cannot have NaNs).
+
```
Comment (DanWaxman) [inline dynestyx/inference/filter_configs.py]: I don't think this is strictly true; it's up to the user to specify transition and observation functions that work okay with NaNs, though.

**Review by DanWaxman — CHANGES_REQUESTED** [review]

File: dynestyx/inference/filter_configs.py (hunk 2 above)
Comment (mattlevine22) [reply dynestyx/inference/filter_configs.py]: okay, I have PF do a warning, and changed the docstring to explain what you said.

File: dynestyx/inference/checkers.py (hunk 1 above)
Comment (mattlevine22) [reply dynestyx/inference/checkers.py]: ok, generalized it (and allow warning, which codex says can't have the same `except` trick)

File: docs/tutorials/gentle_intro/11_missing_observations.ipynb [hunk 3]
```diff
@@ -0,0 +1,797 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c918d8be",
+   "metadata": {},
+   "source": [
+    "# Part 11: Handling Missing Observations with Filters and Smoothers\n",
+    "\n",
+    "Real-world time-series are rarely complete. Sensors drop out, experiments are paused, and some coordinates go missing while others remain observed. This notebook mirrors the missing-observations tutorial, but uses `Filter` and `Smoother` for inference instead of simulator-conditioned latent-state inference.\n",
```
Comment (DanWaxman) [inline docs/tutorials/gentle_intro/11_missing_observations.ipynb]: > Sensors drop out, experiments are paused, and some coordinates go missing while others remain observed.

Feels a bit LLMey :)

File: docs/tutorials/gentle_intro/11_missing_observations.ipynb
Comment (DanWaxman) [inline docs/tutorials/gentle_intro/11_missing_observations.ipynb:1]: We should write something about EnKF working too

File: tests/test_hierarchical_simulator_discretizer_smokes.py [hunk 4]
```diff
@@ -125,8 +124,8 @@ def _plate_continuous_ode_model(
 ):
     with dsx.plate("trajectories", M):
         alpha = numpyro.sample("alpha", dist.Uniform(0.1, 0.8))
-        A_base = jnp.array([[0.0, 0.1], [0.1, 0.8]])
-        A = jnp.repeat(A_base[None], M, axis=0).at[:, 0, 0].set(alpha)
+        A_base = jnp.array([[0.0, 0.1], [-0.05, -0.6]])
+        A = jnp.repeat(A_base[None], M, axis=0).at[:, 0, 0].set(-alpha)
```
Comment (DanWaxman) [inline tests/test_hierarchical_simulator_discretizer_smokes.py:128]: Why change this?

File: tests/test_hierarchical_simulator_discretizer_smokes.py [hunk 5]
```diff
@@ -428,9 +427,9 @@ def _plate_continuous_for_discretizer_model(
 ):
     with dsx.plate("trajectories", M):
         alpha = numpyro.sample("alpha", dist.Uniform(0.1, 0.8))
-        A_base = jnp.array([[0.0, 0.1], [0.1, 0.8]])
-        A = jnp.repeat(A_base[None], M, axis=0).at[:, 0, 0].set(alpha)
-        L = 0.2 * jnp.array([[1.0], [0.5]])
+        A_base = jnp.array([[0.0, 0.1], [-0.05, -0.6]])
+        A = jnp.repeat(A_base[None], M, axis=0).at[:, 0, 0].set(-alpha)
+        L = 0.2 * jnp.eye(2)
```
Comment (DanWaxman) [inline tests/test_hierarchical_simulator_discretizer_smokes.py:432]: Same

**Review by DanWaxman — APPROVED** [review]
Some small things -- will approve, think they should be simple to fix

File: tests/test_hierarchical_simulator_discretizer_smokes.py (hunk 4 above)
Comment (mattlevine22) [reply tests/test_hierarchical_simulator_discretizer_smokes.py:128]: because the previous A was not stable for continuous-time systems, and we sometimes got synthetic data that was NaN.

These NaNs make some of the tests fail ("rightly"), because filter setup does not always support NaN data.

File: tests/test_hierarchical_simulator_discretizer_smokes.py (hunk 5 above)
Comment (mattlevine22) [reply tests/test_hierarchical_simulator_discretizer_smokes.py:432]: see above

Comment (mattlevine22) [reply tests/test_hierarchical_simulator_discretizer_smokes.py:432]: I can change the L back if you want to exercise the padding, but I'd rather just have that be an explicit test if we are worried about padding coverage.

File: docs/tutorials/gentle_intro/11_missing_observations.ipynb
Comment (mattlevine22) [reply docs/tutorials/gentle_intro/11_missing_observations.ipynb:1]: done

File: docs/tutorials/gentle_intro/11_missing_observations.ipynb (hunk 3 above)
Comment (mattlevine22) [reply docs/tutorials/gentle_intro/11_missing_observations.ipynb]: ya ok re-wrote
